zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -539,11 +539,23 @@ function NotebookCtrl($scope, $route, $routeParams, $location, $rootScope,
   $scope.updateNoteName = function(newName) {
     const trimmedNewName = newName.trim();
     if (trimmedNewName.length > 0 && $scope.note.name !== trimmedNewName) {
+      $scope.note.oldName = $scope.note.name;
       $scope.note.name = trimmedNewName;
       websocketMsgSrv.renameNote($scope.note.id, $scope.note.name, true);
     }
   };
 
+  $scope.$on('setNoteMenu', function(event, notes) {
+    $scope.note.oldName = undefined;
+  });
+
+  $scope.$on('errorInfo', function(event, notes) {
+    if ($scope.note.oldName !== undefined) {
+      $scope.note.name = $scope.note.oldName;
+      $scope.note.oldName = undefined;
+    }
+  });
```

Comment:
Couldn't other errors occur while `renameNote()` is in progress?

Reply:
Yes, I can think of 2 alternatives:
- Return a specific error response for the rename action
- Push a `NOTES_INFO` update in the error case to update the UI

What are you thoughts on the 2 approaches?

Reply:
I went with a combined approach:
- Define an explicit exception for this error
- Push a note update message to refresh stale UI